### PR TITLE
Test geom builders

### PR DESCRIPTION
**Pull request by xjjiang**
### Summary

- Created two unit tests: the GASP based geom subsystem builder and the FLOPS based geom subsystem builder.
- Changed the default value of `Aircraft.HorizontalTail.ASPECT_RATIO` to 4.75 in `subsystems/geometry/flops_based/characteristic_lengths.py`
- Changed the default value of `Aircraft.Wing.FOLDED_SPAN` to 25 in `subsystems/geometry/flops_based/non_dimensional_conversion.py`
- Changed the default value of `Aircraft.HorizontalTail.TAPER_RATIO` to 0.352 in `subsystems/mass/flops_based/horizontal_tail.py`
- In `get_parameters()` method of `subsystems/geometry/geometry_builder.py`, excluded `__dict__` from `Aircraft.Nacelle.__dict__` because its value is Nacelle object which can not be handled by mappingproxy.
- Created two unit test files: `test_flops_mass_builder.py` and `test_gasp_mass_builder.py` in `subsystems/geometry/test`.

### Related Issues

- Resolves #

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

**Review by Kenneth-T-Moore — APPROVED** [review]
I'm not too big a fan of changing the openmdao defaults values in some geo components just so the unit test can be run without calling set_input_defaults manually. Still, this is only a problem because some components currently use those defaults. We should come up with a more uniform policy for setting them -- maybe  all 0 or 1, to make it obvious if they were forgotten in the csv.  Until then, I will approve the PR.

**Comment by xjjiang** [comment]
> I'm not too big a fan of changing the openmdao defaults values in some geo components just so the unit test can be run without calling set_input_defaults manually. Still, this is only a problem because some components currently use those defaults. We should come up with a more uniform policy for setting them -- maybe all 0 or 1, to make it obvious if they were forgotten in the csv. Until then, I will approve the PR.

Agree. 
- In testing subsystem builders, there is no set_input_defaults available. Otherwise, I would have used it.
- Software development has a basic principle that code and data should be separated. But We have data everywhere. Since I am sure why we have those non-zero default values, I was worried that some non-zero default values are used somewhere and zero default values somewhere else. This inconsistency is not accepted by the builders.
- I agree that we should be consistent to set all 0 or 1. 
- I will change the defaults of the variables that I touched (including the ones in mass) to zeros and see if It breaks anything.

File: aviary/subsystems/geometry/gasp_based/non_dimensional_conversion.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def setup(self):
         add_aviary_input(self, Aircraft.Wing.SPAN, val=0)
 
         if self.options["aviary_options"].get_val(Aircraft.Wing.FOLD_DIMENSIONAL_LOCATION_SPECIFIED, units='unitless'):
-            add_aviary_input(self, Aircraft.Wing.FOLDED_SPAN, val=0)
+            add_aviary_input(self, Aircraft.Wing.FOLDED_SPAN, val=25)
```
Comment (crecine) [inline aviary/subsystems/geometry/gasp_based/non_dimensional_conversion.py:82]: should probably add units here

File: aviary/subsystems/geometry/test/test_flops_geom_builder.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    That class inherits from TestSubsystemBuilder. So all the test functions are
+    within that inherited class. The setUp() method prepares the class and is run
+    before the test methods; then the test methods are run.
+    """
+
+    def setUp(self):
+        self.subsystem_builder = CoreGeometryBuilder(
+            'core_geometry',
+            BaseMetaData,
+            use_both_geometries=True,
```
Comment (crecine) [inline aviary/subsystems/geometry/test/test_flops_geom_builder.py:28]: should this be False for the non hybrid one?

File: aviary/subsystems/geometry/test/test_gasp_geom_builder.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    That class inherits from TestSubsystemBuilder. So all the test functions are
+    within that inherited class. The setUp() method prepares the class and is run
+    before the test methods; then the test methods are run.
+    """
+
+    def setUp(self):
+        self.subsystem_builder = CoreGeometryBuilder(
+            'core_geometry',
+            BaseMetaData,
+            use_both_geometries=True,
```
Comment (crecine) [inline aviary/subsystems/geometry/test/test_gasp_geom_builder.py:28]: Should this be false?

**Review by crecine — APPROVED** [review]

**Comment by crecine** [comment]
> * In testing subsystem builders, there is no set_input_defaults available. Otherwise, I would have used it.

can we use set_value instead of set_input_default for testing the subsystem builders?

File: aviary/subsystems/geometry/test/test_flops_geom_builder.py (hunk 2 above)
Comment (xjjiang) [reply aviary/subsystems/geometry/test/test_flops_geom_builder.py]: Done. Theoretically, we should test all the scenarios of all the booleans on and off. But each case add twenty more tests. Should we do that?

File: aviary/subsystems/geometry/test/test_gasp_geom_builder.py (hunk 3 above)
Comment (xjjiang) [reply aviary/subsystems/geometry/test/test_gasp_geom_builder.py]: Changed to use_both_geometries=False. Should we test both?

**Comment by xjjiang** [comment]
> > * In testing subsystem builders, there is no set_input_defaults available. Otherwise, I would have used it.
> 
> can we use set_value instead of set_input_default for testing the subsystem builders?

I tried. It does not work.

File: aviary/subsystems/geometry/test/test_flops_geom_builder.py (hunk 2 above)
Comment (crecine) [reply aviary/subsystems/geometry/test/test_flops_geom_builder.py:28]: Maybe I'm misunderstanding what these tests are for, but I thought TestFLOPSGeomBuilder would be testing the pure FLOPS based geometry, TestGASPGeomBuilder would test the pure GASP based geometry, and then TestFLOPSGeomBuilderHybrid and TestGASPGeomBuilderHybrid would be testing when both are used.

Comment (xjjiang) [reply aviary/subsystems/geometry/test/test_flops_geom_builder.py:28]: I tried it but all unit test fail if `use_both_geometries=False`. I rolled it back.

Comment (crecine) [reply aviary/subsystems/geometry/test/test_flops_geom_builder.py:28]: when use_both_geometries is false, `code_origin` needs to be set to either LegacyCode.GASP or LegacyCode.FLOPS

File: aviary/subsystems/geometry/test/test_flops_geom_builder.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            code_origin_to_prioritize=FLOPS)
```
Comment (crecine) [inline aviary/subsystems/geometry/test/test_flops_geom_builder.py:29]:
```suggestion
        self.subsystem_builder = CoreGeometryBuilder(
            'core_geometry',
            BaseMetaData,
            use_both_geometries=False,
            code_origin=FLOPS,
            code_origin_to_prioritize=FLOPS)
```

**Pull request merged**